appcues/src/main/java/com/appcues/analytics/AnalyticsTracker.kt

```diff
@@ -75,6 +75,31 @@ internal class AnalyticsTracker(
         analyticsQueueProcessor.flushAsync()
     }
 
+    // return true if we have a valid session and can track analytics, false if not.
+    private fun checkSession(): Boolean {
+
+        if (sessionMonitor.sessionId == null || sessionMonitor.isExpired) {
+
+            // if no existing session, or expired, start a new one
+            if (sessionMonitor.startNewSession()) {
+                // immediately track the session_started analytic, before any
+                // subsequent analytics for the session are queued
+                val activityRequest = activityBuilder.track(AnalyticsEvent.SessionStarted.eventName, null)
+                updateAnalyticsFlow(EVENT, true, activityRequest)
+                analyticsQueueProcessor.queueThenFlush(activityRequest)
+            } else {
+                // this means no session could be started (no user) and we cannot track
+                // anything - return false to caller
+                return false
+            }
+        } else {
+            // we have a valid session, update its last activity timestamp to push out the timeout
+            sessionMonitor.updateLastActivity()
+        }
+
+        return true
+    }
```

Comment:
this is the key new spot where everything session related is managed - and called by each tracking usage above in this class. This will either start a new session when needed, or bump out the timestamp for the existing one to delay timeout.